modules/perception/obstacle/onboard/lane_post_processing_subnode.cc

```diff
@@ -218,5 +181,69 @@ void LanePostProcessingSubnode::PublishDataAndEvent(
   AINFO << "succeed to publish data and event.";
 }
 
+Status LanePostProcessingSubnode::ProcEvents() {
+  // fusion output subnode only subcribe the fusion subnode
+  CHECK_EQ(sub_meta_events_.size(), 1u) << "only subcribe one event.";
+  const EventMeta &event_meta = sub_meta_events_[0];
+  Event event;
+  event_manager_->Subscribe(event_meta.event_id, &event);
+  ++seq_num_;
+  shared_ptr<SensorObjects> objs;
+  if (!GetSharedData(event, &objs)) {
+    AERROR << "Failed to get shared data. event:" << event.to_string();
+    return Status(ErrorCode::PERCEPTION_ERROR, "Failed to proc events.");
+  }
+
+  cv::Mat lane_map = objs->camera_frame_supplement->lane_map;
+  if (lane_map.empty()) {
+    AERROR << "Get NULL lane_map from camera frame supplement";
+    return Status(ErrorCode::PERCEPTION_ERROR, "Failed to proc events.");
+  }
+
+  LaneObjectsPtr lane_objects(new LaneObjects());
+  CameraLanePostProcessOptions options;
+  options.timestamp = event.timestamp;
+  timestamp_ns_ = event.timestamp * 1e9;
+
+  lane_post_processor_->Process(lane_map, options, &lane_objects);
+  for (size_t i = 0; i < lane_objects->size(); ++i) {
+    (*lane_objects)[i].timestamp = event.timestamp;
+    (*lane_objects)[i].seq_num = seq_num_;
+  }
+  AINFO << "Before publish lane objects, objects num: "
+        << lane_objects->size();
+
+  PublishDataAndEvent(event.timestamp, lane_objects);
+
+  if (publish_) {
+    PublishPerceptionPb(lane_objects);
+  }
+
+  AINFO << "Successfully finished lane post processing";
+  return Status::OK();
+}
+
+void LanePostProcessingSubnode::PublishPerceptionPb(
+    const LaneObjectsPtr &lane_objects) {
+  AINFO << "Lane post-processor publish lane object pb data";
+
+  PerceptionObstacles obstacles;
+
+  // Header
+  common::adapter::AdapterManager::FillPerceptionObstaclesHeader(
+      "perception_obstacle", &obstacles);
+  common::Header *header = obstacles.mutable_header();
+  header->set_lidar_timestamp(0);
+  header->set_camera_timestamp(timestamp_ns_);
+  header->set_radar_timestamp(0);
+
+  // generate lane marker protobuf messages
+  LaneMarkers* lane_markers = obstacles.mutable_lane_marker();
+  LaneObjectsToLaneMarkerProto(*lane_objects, lane_markers);
+
+  common::adapter::AdapterManager::PublishPerceptionObstacles(obstacles);
```

Comment:
... where do we want to publish sensor obstacle?  obstacles are published here with lane marking only, and the camera process subnode is turned off